src/backend/renderer.js

```diff
@@ -774,30 +776,30 @@ export function attach(
     if (existingID !== undefined) {
       return existingID;
     }
-    const id = pendingStringTable.size + 1;
-    pendingStringTable.set(str, id);
+    const stringID = pendingStringTable.size + 1;
+    pendingStringTable.set(str, stringID);
     // The string table total length needs to account
     // both for the string length, and for the array item
     // that contains the length itself. Hence + 1.
     pendingStringTableLength += str.length + 1;
-    return id;
+    return stringID;
```

Comment:
I renamed some `id` variables in this file to say which exactly IDs (i.e. `stringID` or `fiberID`), otherwise it was confusing.

I would further suggest to use `opaque type StringID: number`, `opaque type FiberID: number`, etc. for them (a Flow equivalent of TypeScript branded types) to reliably distinguish the specific types of numbers we deal with around the codebase.

Reply:
`opaque type StringID: number = number` ?

Reply:
> `opaque type StringID: number = number` ?

Yes. This will ensure that this specific number is incompatible with other specific numbers, only with non-specific numbers. The name could be more verbose, i.e. `BridgeProtocolStringID`.

Reply:
I'm still confused by the name "StringID" for a numeric type.

Reply:
I get the concept of using a named type alias. I just think "string id" is a confusing name for this one :grin

Reply:
Yes, oh well English word order is so ambiguous... can be StringTableID but it's not an ID of the table, too. I'd say it's `IdOfStringInBridgeProtocolStringTable` but nobody's going to write this, right?

Reply:
😆 It's an ID for items _in_ the table so I think that would still be a reasonable name.

> I'd say it's `IdOfStringInBridgeProtocolStringTable` but nobody's going to write this, right?

Definitely not! lol